### PR TITLE
chore(server): remove docker-compose and cleanup

### PR DESCRIPTION
**Pull request by fortuna**
In this PR:
- I'm removing Docker Compose. It was a bit of work to figure out the commands, but now the test is more self-contained and it's one less dependency to install.
- Added cache to the npm ci run on Docker build
- Started using fully-qualified image names, so it's not Docker-centric.
- Made the integration_test/test.sh runnable directly. Great for development or testing any image.
- Stopped running a util container. It's not needed because it doesn't run a service.
- Added STOPSIGNAL entries, so we don't need to specify on docker run
- Added a missing dev dependency, minimist.

### CODE REVIEW

File: src/shadowbox/integration_test/test.sh [hunk 1]
```diff
@@ -71,18 +79,64 @@ function ss_arguments_for_user() {
...
 
 function fail() {
   echo FAILED: "$@"
   exit 1
 }
 
+function setup() {
+  shutdown_containers
+
+  podman network create -d bridge "${NET_OPEN}"
```
[the remark this answers is not in the document]
Comment (fortuna) [reply src/shadowbox/integration_test/test.sh:93]: This is standard Docker networking: https://docs.docker.com/network/
There's really nothing special about it.

File: src/shadowbox/integration_test/test.sh [hunk 2]
```diff
@@ -71,18 +79,64 @@ function ss_arguments_for_user() {
...
+  # Use -i to keep the container running.
+  podman run -d --rm -it -p "30555:555" --network "${NET_BLOCKED}" --name "${CLIENT_CONTAINER}" "${CLIENT_IMAGE}"
+}
+
+function shutdown_containers() {
+    podman rm -f -v "${TARGET_CONTAINER}" || true
+    podman rm -f -v "${SHADOWBOX_CONTAINER}" || true
+    podman rm -f -v "${CLIENT_CONTAINER}" || true
+    podman network rm -f "${NET_OPEN}" || true
+    podman network rm -f "${NET_BLOCKED}" || true
```
Comment (sbruens) [inline src/shadowbox/integration_test/test.sh:133]: What I like to do for cleanups is add a consistent tag to all containers, images, and networks. Then you can remove everything by tag, e.g. `docker network rm -f $(docker network ls -q --filter "label=org.getoutline.tag=${TAG}")`. (I assume podman can do the same thing?) I wonder if that could help scale this? Just a thought.

Comment (fortuna) [reply src/shadowbox/integration_test/test.sh:133]: I like the idea, but the integration test should be able to test the image without affecting the image building. So no labels. That could break an existing container, for instance.
However, we can probably use the names the same way. I already have the names with a prefix

**Comment by fortuna** [comment]
This is ready for review now. We're sticking to Docker, but not docker-compose

File: src/shadowbox/integration_test/test.sh (hunk 2 above)
Comment (fortuna) [reply src/shadowbox/integration_test/test.sh]: Oh, I didn't realize we can set labels at run time. I thought it was build time only. We should consider that, but not for this PR.

File: src/shadowbox/integration_test/test.sh [hunk 3]
```diff
@@ -79,10 +86,59 @@ function fail() {
...
+  # Client service.
+  docker build --force-rm -t "${CLIENT_IMAGE}" "$(dirname "$0")/client"
+  # Use -i to keep the container running.
+  docker run -d --rm -it -p "30555:555" --network "${NET_BLOCKED}" --name "${CLIENT_CONTAINER}" "${CLIENT_IMAGE}"
+
+  # Utilities
+  docker build --force-rm -t "${UTIL_IMAGE}" "$(dirname "$0")/util"
+}
+
+function shutdown_containers() {
```
Comment (sbruens) [inline src/shadowbox/integration_test/test.sh:130]: This isn't just shutting down containers, it's also removing them. Maybe rename this function to reflect that or at least leave a comment for people less familiar with what `docker rm -f` does?

**Review by sbruens — APPROVED** [review]

**Pull request merged**